b2/autocomplete_cache.py

```diff
@@ -0,0 +1,163 @@
+######################################################################
+#
+# File: b2/autocomplete_cache.py
+#
+# Copyright 2020 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+from __future__ import annotations
+
+import abc
+import argparse
+import hashlib
+import os
+import pathlib
+import pickle
+from typing import Callable, Iterable
+
+import argcomplete
+
+
+def identity(x):
+    return x
+
+
+class StateTracker(abc.ABC):
+    @abc.abstractmethod
+    def current_state_identifier(self) -> str:
+        raise NotImplementedError()
+
+
+class PickleStore(abc.ABC):
+    @abc.abstractmethod
+    def get_pickle(self, identifier: str) -> bytes | None:
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def set_pickle(self, identifier: str, data: bytes) -> None:
+        raise NotImplementedError()
+
+
+class FileSetStateTrakcer(StateTracker):
+    _files: list[pathlib.Path]
+
+    def __init__(self, files: Iterable[pathlib.Path]) -> None:
+        self._files = list(files)
```

Comment:
```suggestion
        self._files = sorted(files)
```
seems important here

Reply:
ditto